src/components/Tipping/index.tsx

```diff
@@ -0,0 +1,440 @@
+	{ key: 'threeDollar', value: 3 },
+	{ key: 'fiveDollar', value: 5 },
+	{ key: 'tenDollar', value: 10 },
+	{ key: 'fifteenDollar', value: 15 }
+];
+
+const Tipping = ({ className, destinationAddress, open, setOpen, username }: Props) => {
+	const { network } = useNetworkSelector();
+	const { loginWallet, loginAddress, currentTokenPrice } = useUserDetailsSelector();
+	const { api, apiReady } = useApiContext();
+	const [form] = Form.useForm();
+	const [wallet, setWallet] = useState<Wallet>(loginWallet as Wallet);
+	const [address, setAddress] = useState<string>('');
+	const [availableWallets, setAvailableWallets] = useState<any>({});
+	const [accounts, setAccounts] = useState<InjectedTypeWithCouncilBoolean[]>([]);
+	const [loadingStatus, setLoadingStatus] = useState<LoadingStatusType>({ isLoading: true, message: '' });
+	const [isMetamaskWallet, setIsMetamaskWallet] = useState<boolean>(false);
+	const [availableBalance, setAvailableBalance] = useState<BN>(ZERO_BN);
+	const [tipAmount, setTipAmount] = useState<BN>(ZERO_BN);
+	const disable = loadingStatus.isLoading || availableBalance.lte(tipAmount) || !address || tipAmount.eq(ZERO_BN);
+	const [remark, setRemark] = useState<string>('');
+	const [existentialDeposit, setExistentialDeposi] = useState<BN>(ZERO_BN);
+	const unit = chainProperties[network]?.tokenSymbol;
+	const [dollarToTokenBalance, setDollarToTokenBalance] = useState<{ threeDollar: string; fiveDollar: string; tenDollar: string; fifteenDollar: string }>({
+		fifteenDollar: '0',
+		fiveDollar: '0',
+		tenDollar: '0',
+		threeDollar: '0'
+	});
+
+	const handleTipChangeToDollar = (value: number) => {
+		const tip = value / Number(currentTokenPrice || 1);
+		return String(tip);
+	};
+
+	useEffect(() => {
+		if (!network) return;
+		setLoadingStatus({ isLoading: false, message: 'Awaiting for network' });
+		formatBalance.setDefaults({
+			decimals: chainProperties[network].tokenDecimals,
+			unit: chainProperties[network].tokenSymbol
+		});
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [network]);
+
+	useEffect(() => {
+		if (!currentTokenPrice || !currentTokenPrice.length) return;
+		setDollarToTokenBalance({
+			fifteenDollar: handleTipChangeToDollar(15),
+			fiveDollar: handleTipChangeToDollar(5),
+			tenDollar: handleTipChangeToDollar(10),
+			threeDollar: handleTipChangeToDollar(3)
+		});
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [currentTokenPrice]);
+
+	const handleCancel = (e?: any) => {
+		e?.preventDefault();
+		e?.stopPropagation();
+		setTipAmount(ZERO_BN);
+		setRemark('');
+		form.setFieldValue('balance', '');
+		setLoadingStatus({ isLoading: false, message: '' });
+		setOpen(false);
+	};
+
+	const getWallet = () => {
+		const injectedWindow = window as Window & InjectedWindow;
+		setAvailableWallets(injectedWindow.injectedWeb3);
+		setIsMetamaskWallet((injectedWindow as any)?.ethereum?.isMetaMask);
```

Comment:
Remove handling for metamask wallet. First iteration will only support substrate wallets